common/v2/components/WalletUnlock/Web3Provider.tsx

```diff
@@ -44,6 +45,11 @@ class Web3ProviderDecrypt extends Component<Props & ISettingsContext & INetworkC
   public render() {
     const { web3Unlocked, web3ProviderSettings: provider } = this.state;
     const isDefault = provider.id === WalletId.WEB3;
+    AnalyticsService.instance.track(
```

Comment:
1. With React a component can re-renders every time a state variable of a parent changes. We have little control on how many times a component will re-render. This means that in the same session, `.track` can be called multiple times.
2. To avoid this the `.track` call should be made within a hook (on first render) or a lifecycle event.
3. As a general rule, all side effects should be made within a hook. Since `.track` makes an external api call it's a side-effect https://en.wikipedia.org/wiki/Side_effect_(computer_science)

The alternative would be to place the call on click instead of component render.

Reply:
As hooks aren't available in classes, would `componentDidMount` suffice? Or would it be better to place this call on click instead?

Reply:
Clear! Changing this to be on click.